internal/healthchecks/api_check.go

```diff
@@ -49,11 +50,12 @@ func getGCEMetadata() (resourcedetector.GCEResource, error) {
 	return resourcedetector.GCEResource{}, fmt.Errorf("not in GCE")
 }
 
-// monitoringPing reports whether the client's connection to the monitoring service and the
-// authentication configuration are valid. To accomplish this, monitoringPing writes a
-// time series point with empty values to an Ops Agent specific metric.
-// This method mirrors the "(c *Client) Ping" method in "cloud.google.com/go/logging".
-func monitoringPing(ctx context.Context, client monitoring.MetricClient, gceMetadata resourcedetector.GCEResource) error {
+func isInvalidArgumentErr(err error) bool {
+	apiErr, ok := err.(*apierror.APIError)
+	return ok && apiErr.GRPCStatus().Code() == codes.InvalidArgument
```

Comment:
Do we want to consider matching the description as well as the code?

Reply:
`InvalidArgument` can appear in a malformed metric write request. But in this case, the `monitoringPing` is generated to be correct. So it may have value checking for the specific error `One or more points were written more frequently ...` messaged.

The error `APIError` has the method `Reason()` that could match the text. But I'm unsure if that would be maintainable if the message changes in the future. We are also retrying once, so this won't extend long in an edge case.